wagtail/admin/views/pages/listing.py

```diff
@@ -48,10 +55,26 @@ class PageFilterSet(WagtailFilterSet):
         label=_("Date Updated"),
         widget=DateRangePickerWidget,
     )
+    owner = MultipleUserFilter(
+        label=_("Owner"),
+        queryset=(
+            lambda request: get_user_model().objects.filter(
+                pk__in=Page.objects.order_by()
+                .values_list("owner_id", flat=True)
+                .distinct()
+            )
+        ),
+        widget=CheckboxSelectMultiple,
+    )
+    site = SiteFilter(
+        label=_("Site"),
+        queryset=Site.objects.all(),
```

Comment:
We should probably pre-filter this to sites where the user has permissions for at least one page under them. I'm not sure how feasible that is though, so I'm happy to consider it as a future enhancement.